jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/BadResource.java

```diff
@@ -13,29 +13,48 @@
 
 package org.eclipse.jetty.util.resource;
 
-import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URL;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.eclipse.jetty.util.UrlEncoded;
 
 /**
  * Bad Resource.
  *
  * A Resource that is returned for a bade URL.  Acts as a resource
  * that does not exist and throws appropriate exceptions.
  */
-class BadResource extends URLResource
+class BadResource extends Resource
```

Comment:
Yes: `Resource.newResource(URI)` returns a `BadResource` when the resource creation fails with `NoSuchFileException`. This may be a codepath that's now dead, I'll see if I can get rid of it.

Reply:
It's apparently not needed anymore. Let me delete it.